src/actions/attack.cpp

```diff
@@ -144,8 +144,8 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 	poisons = !opp.get_state("unpoisonable") && weapon->bool_ability("poison") && !opp.get_state(unit::STATE_POISONED);
 	backstab_pos = is_attacker && backstab_check(u_loc, opp_loc, units, resources::gameboard->teams());
 	rounds = weapon->get_specials("berserk").highest("value", 1).first;
-	if(weapon->combat_ability("berserk", 1).second) {
-		rounds = weapon->combat_ability("berserk", 1).first;
+	if(!weapon->list_ability("berserk").empty()) {
+		rounds = weapon->list_ability("berserk").highest("value", 1).first;
```

Comment:
you are calling `weapon->list_ability` twice once in the if conditional and once in the body leading to possibly slower code, please refactor  the code so that its only called once. You do the same mistake in many other places.

Reply:
@gfgtdf if this code is correct for you,

int alt_rounds = weapon->list_ability("berserk").highest("value", 1).first;
if(alt_rounds>0 && alt_rounds!=1){
rounds = alt_rounds;
}

Reply:
idk, thje mostz obvous way to do it would be to do it like :
```
auto alt_bersek = weapon->list_ability("berserk");
if(!alt_bersek.empty()) {
		rounds = alt_bersek.highest("value", 1).first;
}
```